test/unit/user_test.rb

```diff
@@ -148,13 +148,50 @@ class UserTest < ActiveSupport::TestCase
       end
   end
 
+
+
+  test 'save user as complete sets status' do
+    crank(:queens_user, barcode: 27777011111111)
+    assert_equal(User::STATUS_LABELS['barcode_pending'], @user.status)
+    @user.save_as_complete!
+    assert_equal(User::STATUS_LABELS['complete'], @user.status)
+  end
+
+
+  test 'save user as pending sets status' do
+    crank(:queens_user, barcode: 27777011111111)
+    @user.save_as_pending!
+    assert_equal(User::STATUS_LABELS['barcode_pending'], @user.status)
+  end
+
+
+  test 'user method send_request_to_patron_creator_service raises
+    an exception on bad parameters' do
+    save_pin = @user.pin
+    @user.pin = nil
+    exception = assert_raise(StandardError) do
+      @user.send_request_to_patron_creator_service(nil)
+    end
+    assert_equal("send_request_to_patron_creator_service called with no pin. User: #{@user.id || 0}", exception.message)
+
+    @user.pin = save_pin
+    save_barcode = @user.barcode
+    @user.barcode = nil
+    exception = assert_raise(StandardError) do
+      @user.send_request_to_patron_creator_service(nil)
+    end
+    assert_equal("send_request_to_patron_creator_service called with no barcode in user object: #{@user.id || 0}", exception.message)
+    @user.barcode = save_barcode
```

Comment:
is this required here?

Reply:
I'm not 100% sure.  The @user object is made in the setup.  I think the setup method is run once, and the @user object is reused between unit tests.  So I'm trying to clean up after myself.  I could also make a new user object directly in the test, but it takes a little bit of extra time, so I try to only do that when necessary.

Reply:
@daryanypl  @user.barcode = save_barcode this line is not required. Because already we did comparison in above line